src/rdkafka_cgrp.c

```diff
@@ -3665,35 +3695,24 @@ void rd_kafka_cgrp_handle_SyncGroup (rd_kafka_cgrp_t *rkcg,
 		rkbuf->rkbuf_rkb = rd_kafka_broker_internal(rkcg->rkcg_rk);
 
         rd_kafka_buf_read_i16(rkbuf, &Version);
-        rd_kafka_buf_read_i32(rkbuf, &TopicCnt);
-
-        if (TopicCnt > 10000) {
-                err = RD_KAFKA_RESP_ERR__BAD_MSG;
+        if (!(assignment = rd_kafka_buf_read_topic_partitions(rkbuf, 0, rd_false)))
                 goto err;
```

Comment:
It needs to go to `err_parse` so that the rkbuf->err (set in buf_read_topic..) will be used, otherwise we'll come to the error label without an error code set.